calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -228,19 +229,41 @@ func unmarshalSliceOfResources(tml []unversioned.TypeMetadata, b []byte) ([]unve
 // Resources.  If the file does not contain any valid Resources this function returns an error.
 func CreateResourcesFromFile(f string) ([]unversioned.Resource, error) {
 	// Load the bytes from file or from stdin.
-	var b []byte
+	var reader io.Reader
 	var err error
-
 	if f == "-" {
-		b, err = ioutil.ReadAll(os.Stdin)
+		reader = os.Stdin
 	} else {
-		b, err = ioutil.ReadFile(f)
+		reader, err = os.Open(f)
 	}
 	if err != nil {
 		return nil, err
 	}
 
-	return createResourcesFromBytes(b)
+	var resources []unversioned.Resource
+	separator := yamlsep.NewYAMLDocumentSeparator(reader)
+	for {
+		b, err := separator.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+
+		r, err := createResourcesFromBytes(b)
+		if err != nil {
+			return nil, err
+		}
+
+		if resources == nil {
```

Comment:
IIRC, you don't need this first branch - you can append to a nil slice which will create the slice for you.